src/ui/components/Library/Team/View/Tests/hooks/useSyncTestIdToUrl.ts

```diff
@@ -0,0 +1,40 @@
+import { useRouter } from "next/router";
+import { Dispatch, SetStateAction, useEffect, useRef } from "react";
+
+import { parseQueryParams } from "ui/components/Library/Team/utils";
+
+export function useSyncTestIdToUrl(
```

Comment:
There's a lot of duplication with the test run version of this. Is there an obvious refactoring to share the implementation code?